bitcoind 27.0 fails during build/install on aarch64-darwin

### Describe the bug

`nix profile install nixpkgs#bitcoind` fails while compiling `bitcoin`

### Steps To Reproduce

Steps to reproduce the behavior:
1. `nix profile install nixpkgs#bitcoind`

### Expected behavior

`bitcoind` is successfully installed.

### Additional context

Here is the output:
```
error: builder for '/nix/store/x3s1ljagibs78b9fpdxa5bdip8cjaqpl-bitcoind-27.0.drv' failed with exit code 2;
       last 10 log lines:
       > 2024-04-22T04:48:47.088571Z [test] [wallet/bdb.cpp:315] [Verify] Using wallet /private/tmp/nix-build-bitcoind-27.0.drv-0/test_common_Bitcoin Core/c9ed87809f413c480df00f73c4737a803456f06b006825530fe30abe43b74d32/receive-requests-0/wallet.dat
       > 2024-04-22T04:48:47.096066Z [test] [wallet/bdb.cpp:161] [Open] BerkeleyEnvironment::Open: LogDir=/private/tmp/nix-build-bitcoind-27.0.drv-0/test_common_Bitcoin Core/c9ed87809f413c480df00f73c4737a803456f06b006825530fe30abe43b74d32/receive-requests-0/database ErrorFile=/private/tmp/nix-build-bitcoind-27.0.drv-0/test_common_Bitcoin Core/c9ed87809f413c480df00f73c4737a803456f06b006825530fe30abe43b74d32/receive-requests-0/db.log
       > 2024-04-22T04:48:47.096397Z [test] [wallet/bdb.cpp:189] [Open] BerkeleyEnvironment::Open: Error -30974 opening database environment: DB_RUNRECOVERY: Fatal error, run database recovery
       > make[3]: *** [Makefile:22947: wallet/test/wallet_tests.cpp.test] Error 1
       > make[3]: Leaving directory '/private/tmp/nix-build-bitcoind-27.0.drv-0/bitcoin-27.0/src'
       > make[2]: *** [Makefile:20856: check-am] Error 2
       > make[2]: Leaving directory '/private/tmp/nix-build-bitcoind-27.0.drv-0/bitcoin-27.0/src'
       > make[1]: *** [Makefile:20521: check-recursive] Error 1
       > make[1]: Leaving directory '/private/tmp/nix-build-bitcoind-27.0.drv-0/bitcoin-27.0/src'
       > make: *** [Makefile:813: check-recursive] Error 1
       For full logs, run 'nix log /nix/store/x3s1ljagibs78b9fpdxa5bdip8cjaqpl-bitcoind-27.0.drv'.
```

### Notify maintainers

@prusnak @fanquake @roconnor 

### Metadata

```
 - system: `"aarch64-darwin"`
 - host os: `Darwin 22.6.0, macOS 13.6.6`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.18.1`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

--- a/pkgs/applications/blockchains/bitcoin/default.nix
+++ b/pkgs/applications/blockchains/bitcoin/default.nix
@@ -13,7 +13,6 @@
 , miniupnpc
 , zeromq
 , zlib
-, db48
 , sqlite
 , qrencode
 , qtbase ? null
@@ -51,7 +50,7 @@ stdenv.mkDerivation rec {
     ++ lib.optionals withGui [ wrapQtAppsHook ];
 
   buildInputs = [ boost libevent miniupnpc zeromq zlib ]
-    ++ lib.optionals withWallet [ db48 sqlite ]
+    ++ lib.optionals withWallet [ sqlite ]
     ++ lib.optionals withGui [ qrencode qtbase qttools ];
 
   postInstall = ''

